src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs

```diff
@@ -142,19 +146,35 @@ public int IndexOf(byte value)
 
         int IndexOfRest(byte value, int firstLength)
         {
-            if (_rest == null) return -1;
-            var index = _rest.IndexOf(value);
+            if (Rest == null) return -1;
```

Comment:
Wouldn't it be better to store the result of `Rest`? It isn't a const now.
```csharp
var rest = Rest;
if (rest == null) return -1;
var index = rest.IndexOf(value);
```

Reply:
Yeah, there are many optimizations like this one that we could/should do. Though, I will do this one now.